vm_trace.c

```diff
@@ -1342,7 +1342,7 @@ void
 rb_hook_list_connect_tracepoint(VALUE target, rb_hook_list_t *list, VALUE tpval, unsigned int target_line)
 {
     rb_tp_t *tp = tpptr(tpval);
-    rb_event_hook_t *hook = alloc_event_hook((rb_event_hook_func_t)tp_call_trace, tp->events, tpval,
+    rb_event_hook_t *hook = alloc_event_hook((rb_event_hook_func_t)tp_call_trace, tp->events & ISEQ_TRACE_EVENTS, tpval,
```

Comment:
Since this is can be used by the public rb_tracepoint_new() in the future, could you leave a comment about how the filtering is done here because rb_tracepoint_new() doesn't support targeting stuff yet and how rb_hook_list_connect_tracepoint() is only used when targeting stuff? Alternatively, maybe there is more specific place to do this filtering.

Reply:
I not sure I follow what you mean.

`rb_hook_list_connect_tracepoint` requires a iseq or bmethod, so it cannot target anything except the events in `ISEQ_TRACE_EVENTS`. The events created in `rb_tracepoint_new` could be either internal or external events, but not both. In the case where the TracePoint created in `rb_tracepoint_new` is enabled with a target, this code applies. Otherwise, this code doesn't apply.

Reply:
I'm a bit surprised that hook list implies iseq or bmethod. Without looking at other code, I expected a global hook list to be a possibility. Which why I suggest leaving a comment.

Reply:
A global hook list is created using `rb_add_event_hook2` when the TracePoint is enabled without a target specified. The function `rb_hook_list_connect_tracepoint` is used only for when there is a target iseq/bmethod. Perhaps the function names could be improved....

Reply:
Okay, I don't want to block this fix over a clarity concern. I can rename and/or add docs separately later. Thanks!